src/main/scala/io/iohk/ethereum/faucet/jsonrpc/FaucetRpcService.scala

```diff
@@ -1,56 +1,59 @@
-import io.iohk.ethereum.mallet.service.RpcClient
-import io.iohk.ethereum.network.p2p.messages.CommonMessages.SignedTransactions.SignedTransactionEnc
-import io.iohk.ethereum.rlp
-import io.iohk.ethereum.utils.{ByteStringUtils, Logger}
+import io.iohk.ethereum.utils.Logger
 import monix.eval.Task
 
-class FaucetRpcService(rpcClient: RpcClient, keyStore: KeyStore, config: FaucetConfig) extends Logger {
+import scala.concurrent.ExecutionContext
 
-  private val wallet = keyStore.unlockAccount(config.walletAddress, config.walletPassword) match {
-    case Right(w) =>
-      log.info(s"wallet unlocked for use in faucet (${config.walletAddress})")
-      w
-    case Left(err) =>
-      throw new RuntimeException(s"Cannot unlock wallet for use in faucet (${config.walletAddress}), because of $err")
-  }
+class FaucetRpcService(config: FaucetConfig)(implicit system: ActorSystem) extends RetrySupport with Logger {
 
-  def sendFunds(sendFundsRequest: SendFundsRequest): ServiceResponse[SendFundsResponse] = {
-    val res = for {
-      nonce <- rpcClient.getNonce(wallet.address)
-      txId <- rpcClient.sendTransaction(prepareTx(sendFundsRequest.address, nonce))
-    } yield txId
-
-    res match {
-      case Right(txId) =>
-        val txIdHex = s"0x${ByteStringUtils.hash2string(txId)}"
-        log.info(s"Sending ${config.txValue} ETC to ${sendFundsRequest.address} in tx: $txIdHex.")
-        Task.now(Right(SendFundsResponse(txId)))
-
-      case Left(err) =>
-        log.error(s"An error occurred while using faucet: $err")
-        Task.now(Left(JsonRpcError.InternalError))
-    }
-  }
+  implicit val scheduler = system.scheduler
+  val actorSelected = system.actorSelection(s"user/${FaucetSupervisor.name}/${FaucetHandler.name}")
+  val attempts = config.supervisor.attempts
+  val delay = config.supervisor.delay
+
+  lazy val handlerTimeout: Timeout = Timeout(config.handlerTimeout)
+  implicit lazy val actorTimeout: Timeout = Timeout(config.responseTimeout)
+  implicit val ec: ExecutionContext = system.dispatcher
+
+  private def faucetHandler(): Task[ActorRef] =
+    Task.deferFuture(retry(() => actorSelected.resolveOne(handlerTimeout.duration), attempts, delay))
 
-  private def prepareTx(targetAddress: Address, nonce: BigInt): ByteString = {
-    val transaction =
-      Transaction(nonce, config.txGasPrice, config.txGasLimit, Some(targetAddress), config.txValue, ByteString())
+  def sendFunds(sendFundsRequest: SendFundsRequest): ServiceResponse[SendFundsResponse] =
+    faucetHandler().flatMap(handler =>
+      handler
+        .askFor[Any](FaucetHandlerMsg.SendFunds(sendFundsRequest.address))
+        .map(handleSendFundsResponse orElse handleErrors)
+    )
 
-    val stx = wallet.signTx(transaction, None)
-    ByteString(rlp.encode(stx.tx.toRLPEncodable))
+  def status(statusRequest: StatusRequest): ServiceResponse[StatusResponse] =
+    faucetHandler()
+      .flatMap(handler => handler.askFor[Any](FaucetHandlerMsg.Status))
```

Comment:
what happens if we can't reach the actor? i mean in terms of response...

Reply:
Pending task: ETCM-396